compiler/modules/types/list.bal

```diff
@@ -270,15 +270,16 @@ function listInhabited(Context cx, FixedLengthArray members, SemType rest, Conju
         // SemType d1 = diff(s[1], t[1]);
         // return !isEmpty(cx, d1) &&  tupleInhabited(cx, [s[0], d1], neg.rest);
         // We can generalize this to tuples of arbitrary length.
-        foreach int i in 0 ..< len {
+        int distinctMemberCount = members.initial.length() == 0 ? 0 : int:max(members.initial.length(), nt.members.initial.length());
```

Comment:
You have two uses of members.initial.length() here.

I think I would call the variable `maxInitialLength` and write it like this:

```
int maxInitialLength = members.initial.length();
if maxInitialLength != 0 {
    maxInitialLength = int:max(maxInitialLength, nt.members.initial.length());
}
```